extensions/ipynb/src/ipynbMain.ts

```diff
@@ -41,6 +41,26 @@ export function activate(context: vscode.ExtensionContext) {
 			});
 			return vscode.workspace.applyEdit(edit);
 		},
+		setKernelSpecAndLanguageInfo: async (resource: vscode.Uri, kernelspec: unknown, language_info: unknown): Promise<boolean> => {
```

Comment:
Can `languageInfo` be an optional argument to `setKernelSpec` instead of its own function?

Reply:
Done. I'm not entirely sure, though, whether the way I'm construction the `custom` dictionary now will work... In particular, just not entirely sure whether `...(language_info ? { language_info: language_info } : {})` does what I think it does.

Reply:
@mjbvz @davidanthoff  Why not change the method name completely and pass the entire metadata?
I.e. as follows:

```js
setNotebookMetaata(resoruce: vscode.Uri, metadata: Partial<nbformat.INotbeookMetadata>): Promise<boolean> {
...
}
```

After all, the requirement is to provide an API to set the metadata. and if we end up with more items in the notebook metadata the new method will be future proof.
Today there are at least 3 properties in the metadata (kernelSpec, language_info and orig_nbformat).
I propose changing the method to providing a partial of the metadata.
This will also allow others to add custom metadata into the notebook (after all the ipynb spec allows for that).

```typescript
    /**
     * The default metadata for the notebook.
     */
    interface INotebookMetadata extends JSONObject {
        kernelspec?: IKernelspecMetadata;
        language_info?: ILanguageInfoMetadata;
        orig_nbformat: number;
    }
```

Reply:
Yes, that sounds like a good idea to me. Although the most important objective from our end is to get something into 1.60.1, I think :) @DonJayamanne do you want to take over this PR? You're probably in the best position to do the right thing here.